src/internal/connector/onedrive/collection.go

```diff
@@ -135,6 +146,7 @@ func NewCollection(
 	statusUpdater support.StatusUpdater,
 	source driveSource,
 	ctrlOpts control.Options,
+	colKind collectionScope,
```

Comment:
Nit: naming consistency.  Admittedly, I think I'd prefer `type` just so we don't overload or confuse terminology from selector scopes.  That's minor, though.
```suggestion
	colScope collectionScope,
```

Reply:
`type` would be ideal, but unfortunately we can't use that as it is a reserved keyword.